### PR TITLE
add users/cursor keyset_pagination endpoint

### PR DESCRIPTION
**Pull request by andreaskoepf**

### CODE REVIEW

**Review by notmd — APPROVED** [review]
Very clean, thank you.

File: backend/oasst_backend/api/v1/messages.py [hunk 1]
```diff
@@ -16,38 +16,90 @@
...
+    max_count: Optional[int] = Query(10, gt=0, le=1000),
+    desc: Optional[bool] = False,
+    api_client: ApiClient = Depends(deps.get_api_client),
+    db: Session = Depends(deps.get_db),
+):
+    def split_cursor(x: str | None) -> tuple[datetime.datetime, UUID]:
+        if not x:
+            return None, None
+        f = x.split("$", maxsplit=1)
+        if len(f) == 2 and len(f[0]) == 36:
```
Comment (yk) [inline backend/oasst_backend/api/v1/messages.py:69]: other than `36`, maybe `len("some_constant")` or a comment why 36

File: backend/oasst_backend/api/v1/users.py [hunk 2]
```diff
@@ -70,6 +72,71 @@ def get_users_ordered_by_display_name(
...
+    gt: Optional[str] = None,
+    sort_key: Optional[str] = Query("username", max_length=32),
+    max_count: Optional[int] = Query(100, gt=0, le=10000),
+    api_client_id: Optional[UUID] = None,
+    search_text: Optional[str] = None,
+    auth_method: Optional[str] = None,
+    api_client: ApiClient = Depends(deps.get_api_client),
+    db: Session = Depends(deps.get_db),
+):
+    def split_cursor(x: str | None) -> tuple[str, UUID]:
```
Comment (yk) [inline backend/oasst_backend/api/v1/users.py:87]: duplicate code?

**Review by yk — APPROVED** [review]
thank you!

File: backend/oasst_backend/api/v1/users.py (hunk 2 above)
Comment (andreaskoepf) [reply backend/oasst_backend/api/v1/users.py:87]: one splits into str, uuid, the other in datetime, uuid ... datetime parsing is the part where the two functions differ .. I can give them different names to highlight this.

File: backend/oasst_backend/api/v1/messages.py (hunk 1 above)
Comment (andreaskoepf) [reply backend/oasst_backend/api/v1/messages.py:69]: ok, 36 is the length of a full UUID converted with str() (with dashes) ... I could also validate with a regex ..

**Pull request merged**